🐛 fix : 웹소켓 라우팅 활성화

BE12-OZ opened the pull request:
## ✅ PR 요약
- 관련 이슈 번호: 
- 작업 요약:
- 웹소켓 라우팅 활성화 관련 누락된 데이터 작업

## 📄 상세 내용
- [x] 웹소켓 라우팅 활성화 관련 누락된 데이터 작업
- config/asgi.py는 HTTP 요청만 처리하도록 되어 있어, WebSocket 연결 요청(ws:// 프로토콜)을 거부
- config/asgi.py에 ProtocolTypeRouter 설정 추가


## 📸 스크린샷 (선택)

## 📝 기타 참고 사항

## 🧪 PR Checklist
- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

BE12-OZ committed:
feat: 웹소켓 라우팅 활성화

BE12-OZ committed:
feat: 웹소켓 라우팅 활성화 작업 진행

Meoyoug commented on config/asgi.py:
```diff
@@ -1,16 +1,18 @@
-
 import os
 
+from channels.auth import AuthMiddlewareStack
```
이거 사용하시면 JWT Authentication안되지않나요

BE12-OZ committed:
feat(chat): 웹소켓 연결에 JWT 인증 구현

채팅 애플리케이션의 웹소켓 연결에 JWT 인증을 도입합니다.

- 를 생성하여 웹소켓 스코프에 대한 JWT 토큰 유효성 검사 및 사용자 인증을 처리합니다.
- 를 수정하여 새로운 를 웹소켓 프로토콜 라우터에 통합하고, 기본 을 대체합니다.
-  및 에서 Django 앱 레지스트리가 완전히 초기화된 후에 Django 종속 모듈을 가져오도록 하여  오류를 해결했습니다.
-  타입 검사 오류를 해결하기 위해 에 타입 힌트 및 명시적 캐스트를 추가했습니다.

이 변경으로 웹소켓 연결이 쿼리 파라미터로 전달된 JWT 토큰을 사용하여 인증될 수 있도록 하여, 프로젝트의 기존 JWT 인증 체계와 일치시킵니다.
